uniflow/model/model.py

```diff
@@ -53,10 +51,12 @@ def _serialize(self, data: List[Context]) -> List[str]:
         output = []
         for d in data:
             guided_prompt_template = copy.deepcopy(self._guided_prompt_template)
+            if isinstance(d, Context):
+                d = d.model_dump()
             if "examples" in guided_prompt_template:
-                guided_prompt_template["examples"].append(d.model_dump())
+                guided_prompt_template["examples"].append(d)
```

Comment:
You should not access guided_prompt_template through it dict interface. I assume it is in pydantic and you should access it through .examples?

You should check all other places to update as well to avoid access pydantic class as a dict by instead a data class.